sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ListStateImpl.scala

```diff
@@ -40,6 +40,7 @@ class ListStateImpl[S](
      stateName: String,
      keyExprEnc: ExpressionEncoder[Any],
      valEncoder: Encoder[S],
+     avroEnc: Option[AvroEncoderSpec],
```

Comment:
nit: param comment is missing for this

Also, is there a reason you don't want to default it to `None`? So you don't have to manually pass in None for all the current usage